drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+        ),
+    ):
+        return {"address": hex(address)}
+    elif isinstance(address, capa.features.address.DNTokenAddress):
+        return {"token": address.token}
+    elif isinstance(address, capa.features.address.DNTokenOffsetAddress):
+        return {**format_capa_address(address.token), "offset": address.offset}
+    elif isinstance(address, capa.features.address.ProcessAddress):
+        return {"ppid": address.ppid, "pid": address.pid}
+    elif isinstance(address, capa.features.address.ThreadAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.process)}
+    elif isinstance(address, capa.features.address.DynamicCallAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.thread)}
+    elif isinstance(address, capa.features.address._NoAddress):
+        # empty address
+        return {}
+    else:
+        logger.debug("Encountered unknown address type: %s", type(address))
+        return {"address": address}
+
+
+def construct_ttp_block(
+    rule: capa.rules.Rule, addresses: capa.features.address
+) -> Dict[str, Any]:
+    name = rule.name.split("/")[0]
+    mbc = rule.meta.get("mbc", None)
+    attck = rule.meta.get("att&ck", None)
+    occurrences = [format_capa_address(address=address) for address, _ in addresses]
+    occurrences = [
+        dict(t) for t in {tuple(d.items()) for d in occurrences}
+    ]  # remove duplicate addresses
+
+    ttp_block = dict()
+    ttp_block.update({"name": name})
+    ttp_block.update({"mbc": mbc} if mbc else {})
+    ttp_block.update({"att&ck": attck} if attck else {})
+    ttp_block.update({"occurrences": occurrences})
+
+    return ttp_block
+
+
+def construct_ttp_blocks(
+    rules: capa.rules.RuleSet,
+    capabilities_per_file: List[Tuple[Path, capa.engine.MatchResults, Any]],
+    filter_function=lambda rule: rule,
+) -> Iterator[Dict[str, Any]]:
+
+    # construct a ttp block for each extracted capability
+    for _, capabilities in capabilities_per_file:
+        for name, addresses in capabilities.items():
+            if filter_function(rules[name]):
+                yield construct_ttp_block(rules[name], addresses)
+
+
+def capa_analysis(analysis_dir: Path) -> None:
+    # check and prepare the rules folder
+    if not check_rules_directory_exist(capa_rules_dir):
+        # in case of a missing/empty rules folder, clone the official capa rules
+        logger.exception("capa rules directory is empty or non-existant")
+
+    # get rules and filter them
+    rules = get_rules([capa_rules_dir])
+    rules = filter_rules(
+        rules, filter_function=lambda rule: rule.meta.get("att&ck", None)
+    )  # select only rules with an att&ck entry specified
+
+    # get malware-related pids if requested by configuration
+    malware_pids = None
+    if analyze_malware_pids_only:
+        malware_pids = list(
```

Comment:
This will crash if get_malware_processes return None. But as per comment at get_malware_processes, it should just always return a value.

Reply:
yeah this is fixed now.